src/commons/eventsourcing/wal.rs

```diff
@@ -285,6 +314,11 @@ impl<T: WalSupport> WalStore<T> {
     /// This is a separate function because serializing a large instance can
     /// be expensive.
     pub fn update_snapshot(&self, handle: &MyHandle, archive: bool) -> WalStoreResult<()> {
+        // Note: We do not need to keep a read lock for the instance when we update the snapshot.
```

Comment:
I don't understand how it is okay not to lock here.

Reply:
Will improve the comment then.. but essentially..

This updates the latest snapshot file on disk (or well in the KV store). It is okay to do this with a copy of the data, even if that copy becomes outdated as the snapshot is written. All changes are stored in their own write-ahead-log (wal) file. So, even if we end up saving a now outdated snapshot, we will have all changes after it.

If we would keep the lock, then this would mean that write operations would have to wait until the full snapshot file is written. For a large repository (1000s of CAs and ROAs) this can take a while (up to 30 seconds on my fast machine for a 250 MB repo) because the serialisation is expensive. I think it's better not to block the system for this long if it's not needed.

Reply:
updated comment